components/server/ee/src/user/stripe-service.ts

```diff
@@ -58,9 +64,7 @@ export class StripeService {
             email: User.getPrimaryEmail(user),
             name: User.getName(user),
             metadata: {
-                // userId is deprecated, use attributionId where possible
-                userId: user.id,
-                attributionId: AttributionId.render({ kind: "user", userId: user.id }),
+                ATTRIBUTION_ID_METADATA_KEY: AttributionId.render({ kind: "user", userId: user.id }),
```

Comment:
I don't think this is going to work (you're creating an object key that's literally `"ATTRIBUTION_ID_METADATA_KEY"`).

If you really want to make `ATTRIBUTION_ID_METADATA_KEY` configurable with a global const in this file, you'd want to first create a `const customerSpec = { email, name, metadata: {} }` and then set `customerSpec[ATTRIBUTION_ID_METADATA_KEY] = ...`.

Reply:
Revert is [here](https://github.com/gitpod-io/gitpod/pull/12765). Thanks for spotting this. I applied Go style thinking it would use it as a key, not as a literal..